packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -18,14 +22,39 @@ export default createRule<Options, MessageIds>({
     messages: {
       noVarReqs: 'Require statement not part of import statement.',
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allow: {
+            type: 'array',
+            items: { type: 'string' },
+            description: 'Patterns of import paths to allow requiring from.',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
   },
-  defaultOptions: [],
-  create(context) {
+  defaultOptions: [{ allow: [] }],
+  create(context, options) {
+    const allowPatterns = options[0].allow.map(
+      pattern => new RegExp(pattern, 'u'),
+    );
+    function isImportPathAllowed(importPath: string): boolean {
+      return allowPatterns.some(pattern => importPath.match(pattern));
+    }
     return {
       'CallExpression[callee.name="require"]'(
         node: TSESTree.CallExpression,
       ): void {
+        if (
+          node.arguments[0]?.type === AST_NODE_TYPES.Literal &&
+          typeof node.arguments[0].value === 'string' &&
+          isImportPathAllowed(node.arguments[0].value)
```

Comment:
[Bug] I think this should use `getStaticStringValue` or an equivalent tailored just to strings. Then it would support template literals.

I did a quick search for `type === AST_NODE_TYPES.Literal` and found a few files `Referencer.ts`, `prefer-regexp-exec.ts`, `prefer-reduce-type-parameter.ts`, `padding-line-between-statements.ts`, `no-restricted-imports.ts`...) that do similar _"get the value if it's a string literal"_ logic. Maybe good to make a new utility?

Thoughts?

Reply:
#8186. I wanted to get this PR merged 😄 and I don't think template literals in `require()` calls are pretty rare.